### PR TITLE
Add Gamepad class, and necessary support classes for testing.

### PR DESCRIPTION
**Pull request by JohnGilb**

### CODE REVIEW

File: src/xbot/common/controls/sensors/Gamepad.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package xbot.common.controls.sensors;
+
+import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj.Joystick;
+import xbot.common.controls.sensors.wpi_adapters.GamepadJoystickWpiAdapter;
+import xbot.common.controls.sensors.wpi_adapters.GamepadJoystickWpiAdapter.GamepadComponent;
+
+public class Gamepad implements XGamepad {
+    
+    GenericHID internalHID;
```
Comment (aschokking) [inline src/xbot/common/controls/sensors/Gamepad.java]: might as well be final too?

File: src/xbot/common/controls/sensors/wpi_adapters/GamepadJoystickWpiAdapter.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        RightJoystick,
+        LeftTrigger,
+        RightTrigger,
+    }
+    
+    public GamepadJoystickWpiAdapter(GenericHID internalSharedHID, GamepadComponent side) {
+        this.internalSharedHID = internalSharedHID;
+        this.side = side;
+    }
+    // standard gamepad axis X 0, Y1
```
Comment (aschokking) [inline src/xbot/common/controls/sensors/wpi_adapters/GamepadJoystickWpiAdapter.java]: maybe plop this info a the top in a /* */ comment?

File: src/xbot/common/controls/sensors/Gamepad.java (hunk 1 above)
Comment (JohnGilb) [inline src/xbot/common/controls/sensors/Gamepad.java]: Good catch

File: src/xbot/common/controls/sensors/wpi_adapters/GamepadJoystickWpiAdapter.java (hunk 2 above)
Comment (JohnGilb) [inline src/xbot/common/controls/sensors/wpi_adapters/GamepadJoystickWpiAdapter.java]: Done, and formatted prettily

**Pull request merged**